### ISSUE TITLE
Compiling with GHC 8.0.1-rc4

### ISSUE THREAD

**Issue by davean** [bjornbm/numtype-dk#12]
The following patch fixes a namespace issue:

```
diff --git a/Numeric/NumType/DK/Integers.hs b/Numeric/NumType/DK/Integers.hs
index 02fcd32..612bef5 100644
--- a/Numeric/NumType/DK/Integers.hs
+++ b/Numeric/NumType/DK/Integers.hs
@@ -54,8 +54,7 @@ import qualified Prelude
 
 #if MIN_VERSION_base(4, 8, 0)
 -- Use @Nat@s from @GHC.TypeLits@.
-import GHC.TypeLits hiding ((+)(), (*)(), (-)(), (^)())
-import qualified GHC.TypeLits as N
+import qualified GHC.TypeLits as TN
 
 type Z  = 0  -- GHC.TypeLits
 type N1 = 1  -- GHC.TypeLits
@@ -76,14 +75,14 @@ infixl 6  +, -
 -- Natural numbers
 -- ===============
 
-type family NatPred (n::Nat) :: Nat where NatPred n = n N.- N1
-type family NatSucc (n::Nat) :: Nat where NatSucc n = n N.+ N1
+type family NatPred (n::TN.Nat) :: TN.Nat where NatPred n = n TN.- N1
+type family NatSucc (n::TN.Nat) :: TN.Nat where NatSucc n = n TN.+ N1
 
 
 -- Integers
 -- ========
 
-data TypeInt = Neg10Minus Nat  -- 10, 11, 12, 13, ...
+data TypeInt = Neg10Minus TN.Nat  -- 10, 11, 12, 13, ...
              | Neg9
              | Neg8
              | Neg7
@@ -103,7 +102,7 @@ data TypeInt = Neg10Minus Nat  -- 10, 11, 12, 13, ...
              | Pos7
              | Pos8
              | Pos9
-             | Pos10Plus Nat  -- -10, -11, -12, -13, ...
+             | Pos10Plus TN.Nat  -- -10, -11, -12, -13, ...
 
 
 -- Unary operations
```

**Comment by davean** [comment]
If you prefer, a repo is available at http://code.xkrd.net/davean/numtype-dk.git

**Comment by dmcclean** [comment]
The proposed patch breaks our support for GHC 7.8, which was based on a conditional compilation of imports which ended up importing names that worked for 7.8 or 7.10.

See the build log at https://travis-ci.org/dmcclean/numtype-dk/jobs/128744642

Since I don't have a local install of the GHC 8 release candidate yet it's kind of tough for me to test, but I'll see what I can do about making a version that keeps your change but also still works on 7.8.

**Comment by dmcclean** [comment]
This version is now available on hackage as 0.5.0.1.
